packages/legacy/core/App/screens/Chat.tsx

```diff
@@ -54,12 +54,28 @@ const Chat: React.FC<ChatProps> = ({ route }) => {
   const basicMessages = useBasicMessagesByConnectionId(connectionId)
   const credentials = useCredentialsByConnectionId(connectionId)
   const proofs = useProofsByConnectionId(connectionId)
-  const theirLabel = useMemo(() => connection?.theirLabel || connection?.id || '', [connection])
+  const isFocused = useIsFocused()
   const { assertConnectedNetwork, silentAssertConnectedNetwork } = useNetwork()
   const [messages, setMessages] = useState<Array<ExtendedChatMessage>>([])
   const [showActionSlider, setShowActionSlider] = useState(false)
   const { ChatTheme: theme, Assets } = useTheme()
   const { ColorPallet } = useTheme()
+  const [theirLabel, setTheirLabel] = useState(
+    (connection?.id && store.preferences.alternateContactNames[connection.id]) ||
```

Comment:
This check is done several times, here, `ContactDetails.tsx`, `ProofDetails.tsx`, and maybe others. Wondering if it should be simplified to a fn? Maybe something similar in `ProofRequest.tsx`?

Reply:
Done, sorry that was a bit lazy of me. Added tests for the helper as well